### DISCUSSION TITLE
How to Replace deployment using SDK

### DISCUSSION THREAD

**Discussion by fzwu** [kubernetes-client/csharp#1069] [Q&A] [answered]
Hi,

I am using this C# SDK to replace deployment. I used  AppsV1.ReplaceNamespacedDeploymentAsync method  but  response  Operation returned an invalid status code 'UnprocessableEntity'  .  I don't know why and What's wrong with my V1Deployment, 

this is my V1Deployment Entity,  
{"ApiVersion":"apps/v1","Kind":"Deployment","Metadata":{"Annotations":{"reloader.stakater.com/auto":"true"},"Name":"id-service","NamespaceProperty":"eworldcloud"},"Spec":{"Replicas":4,"Selector":{"MatchLabels":{"app":"id-service"}},"Strategy":{"RollingUpdate":{"MaxSurge":{"Value":"3"},"MaxUnavailable":{"Value":"1"}},"Type":"RollingUpdate"},"Template":{"Metadata":{"Labels":{"app":"id-service"}},"Spec":{"Containers":[{"Env":[{"Name":"MachineIdUrl","ValueFrom":{"ConfigMapKeyRef":{"Key":"machineIdUrl","Name":"configmap-eworldcloud"}}},{"Name":"UnitTestEnabled","ValueFrom":{"ConfigMapKeyRef":{"Key":"unitTestEnabled","Name":"configmap-eworldcloud"}}},{"Name":"DistributedTracingEnabled","ValueFrom":{"ConfigMapKeyRef":{"Key":"distributedTracingEnabled ","Name":"configmap-eworldcloud"}}},{"Name":"LogLevel","ValueFrom":{"ConfigMapKeyRef":{"Key":"logLevel","Name":"configmap-eworldcloud"}}},{"Name":"LogKind","ValueFrom":{"ConfigMapKeyRef":{"Key":"logKind","Name":"configmap-eworldcloud"}}},{"Name":"Plat","ValueFrom":{"ConfigMapKeyRef":{"Key":"plat","Name":"configmap-eworldcloud"}}},{"Name":"MaxCallbackTimespan ","Value":"10"},{"Name":"usezookeeper  ","Value":"1"},{"Name":"ASPNETCORE_URLS  ","Value":"http://*:8001"},{"Name":"IsCanary  ","Value":"0"}],"Image":"registry.mtywcloud.com/development/id-service:v1.0.18.72","ImagePullPolicy":"Always","LivenessProbe":{"FailureThreshold":3,"HttpGet":{"Path":"/actuator/health/liveness","Port":{"Value":"8001"}},"InitialDelaySeconds":15,"PeriodSeconds":15,"TimeoutSeconds":60},"Name":"id-service","Ports":[{"ContainerPort":8001}],"ReadinessProbe":{"FailureThreshold":3,"HttpGet":{"Path":"/actuator/health/readiness","Port":{"Value":"8001"}},"InitialDelaySeconds":15,"PeriodSeconds":15,"TimeoutSeconds":60},"Resources":{"Limits":{"cpu":{"Format":1,"Value":"1000m"},"memory":{"Format":1,"Value":"1024Mi"}},"Requests":{"cpu":{"Format":1,"Value":"100m"},"memory":{"Format":1,"Value":"400Mi"}}}}],"ImagePullSecrets":[{"Name":"dev-harbor-secret"}],"ServiceAccount":"eworld-eworldcloud"}}}}

Thanks

**Comment by tg123** [comment, marked as answer]
This is likely caused by your request is not valid
you may want to catch and inspect the httpresponse of the ex, there the server would tell you the detail

**Reply by fzwu** [reply]
Exception is    在 k8s.Kubernetes.<SendRequestRaw>d__39.MoveNext()
   在 k8s.AbstractKubernetes.<k8s-IAppsV1Operations-CreateNamespacedDeploymentWithHttpMessagesAsync>d__276.MoveNext()
   在 k8s.AppsV1OperationsExtensions.<CreateNamespacedDeploymentAsync>d__47.MoveNext()

**Reply by fzwu** [reply]
I've solved the replacement.First get the existing Deployment，then update the value, use the Entity which i got from ListNamespacedDeploymentAsync  that is will be OK.
But when i want to create Deployment , i used AppsV1.CreateNamespacedDeploymentAsync  method, it's response"  Operation returned an invalid status code 'UnprocessableEntity'   "
StachTrace:
   在 k8s.Kubernetes.<SendRequestRaw>d__39.MoveNext()
   在 k8s.AbstractKubernetes.<k8s-IAppsV1Operations-CreateNamespacedDeploymentWithHttpMessagesAsync>d__276.MoveNext()
   在 k8s.AppsV1OperationsExtensions.<CreateNamespacedDeploymentAsync>d__47.MoveNext()

**Reply by fzwu** [reply]
I've solved,it's because the Key of the environment variable does not match
